### PR TITLE
[FLOC-2757] Release Flocker 1.1.0.dev2

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-2757

News and What's New changes are based on the aborted dev1 release in https://github.com/ClusterHQ/flocker/pull/1779 with a few additions.

### CODE REVIEW

File: docs/releasenotes/index.rst [hunk 1]
```diff
@@ -10,6 +10,14 @@ You can learn more about where we might be going with future releases by:
...
 * Visiting our GitHub repository at https://github.com/ClusterHQ/flocker.
 
+Next Release
+============
+
+* ``flocker-deploy`` supports specification of the pathnames of certificate and key files.
+  See :ref:`flocker-deploy-authentication`.
+* The agent configuration file allows specification of a CA certificate for OpenStack HTTPS verification.
+  See :ref:`openstack-dataset-backend`.
+
```
Comment (itamarst) [inline docs/releasenotes/index.rst]: I feel like rsyslogd and private registeries merit mention here.

**Comment by wallrj** [comment]
@itamarst wrote:
> I feel like rsyslogd and private registeries merit mention here.

I added those to the release notes.

@jongiddy will you take a quick look at this branch?

**Comment by jongiddy** [comment]
This looks good. Please continue when the Vagrant tutorial box test completes OK.

**Comment by jml** [comment]
Just came across this while looking for tickets to review. Should it be marked as accepted?

**Comment by wallrj** [comment]
I encountered a problem with the `publish-artifacts` script (https://clusterhq.atlassian.net/browse/FLOC-2768), but having fixed that the rest of the release process went according to plan.

Merging.

**Pull request merged**